src/library_syscall.js

```diff
@@ -871,8 +871,9 @@ var SyscallsLibrary = {
     var buf = SYSCALLS.get(), size = SYSCALLS.get();
     if (size === 0) return -ERRNO_CODES.EINVAL;
     var cwd = FS.cwd();
-    if (size < cwd.length + 1) return -ERRNO_CODES.ERANGE;
-    writeAsciiToMemory(cwd, buf);
+    var cwdLengthInBytes = lengthBytesUTF8(cwd);
+    if (size < cwdLengthInBytes + 1) return -ERRNO_CODES.ERANGE;
+    stringToUTF8(cwd, buf, cwdLengthInBytes + 1);
```

Comment:
Well, the condition above guarantees that `size >= cwdLengthInBytes + 1`, so there is no error here. But I can change it to `size` if you think that the code will be more clear.